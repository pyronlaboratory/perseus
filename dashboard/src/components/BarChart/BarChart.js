import React from 'react';
import { Bar, defaults } from 'react-chartjs-2';

defaults.animation = false;

const BarChart = (props) => {
    let labels = [];
    let distributionData = [];
    
    props.data.map((item) => (
      labels.push(item.month)
    ));
  
    props.data.map((item) => (
      distributionData.push(item.total_number_of_users)
    ));
  
    
    const chartData = {
  
      labels: labels,
      datasets: [
        {
          label: '# user attempt per month distribution',
          data: distributionData,
          backgroundColor: ['#080A0E', '#082032', '#2C394B', '#334756', '#3C5467', '#445E74', '#4B6981', '#53738D', '#5B7E9A', '#6588A4', '#7292AC', '#7E9CB4'],
          borderColor: ['#080A0E', '#082032', '#2C394B', '#334756', '#3C5467', '#445E74', '#4B6981', '#53738D', '#5B7E9A', '#6588A4', '#7292AC', '#7E9CB4'],
          borderWidth: 1,
        },
      ],
    }
  
    return (
        <Bar 
            data={chartData}
            options={{
                plugins: {
                    legend: {
                        position: 'bottom',
                        align: 'center',
                        display: false
                    }
                }
            }}
        />
    );
}

export default BarChart