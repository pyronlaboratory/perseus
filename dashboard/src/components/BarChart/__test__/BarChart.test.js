import React from 'react';
import ReactDOM from 'react-dom';

// react testing
import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';

import BarChart from './../BarChart';

afterEach(cleanup);

describe('bar chart component', () => {
    it("renders without crashing", () => {
        const mockData = {
            data: [
                {"month": "string", "total_number_of_users": 0},
                {"month": "string", "total_number_of_users": 0}
            ],
            options: "chart options"
        }
        const div = document.createElement("div");
        ReactDOM.render(<BarChart data={mockData.data} options={mockData.options}/>, div);
    })

    it("matches snapshot", () => {
        const mockData = {
            data: [
                {"month": "string", "total_number_of_users": 0},
                {"month": "string", "total_number_of_users": 0}
            ],
            options: "chart options"
        }

        const tree = renderer.create(<BarChart data={mockData.data} options={mockData.options} />).toJSON();
        expect(tree).toMatchSnapshot();
    })
})