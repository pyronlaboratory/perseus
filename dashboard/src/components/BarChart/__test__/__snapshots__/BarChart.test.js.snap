// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bar chart component matches snapshot 1`] = `
<canvas
  data-testid="canvas"
  height={150}
  onClick={[Function]}
  role="img"
  width={300}
/>
`;
